src/components/views/messages/DateSeparator.tsx

```diff
@@ -132,28 +136,106 @@ export default class DateSeparator extends React.Component<IProps, IState> {
                     `found ${eventId} (${originServerTs}) for timestamp=${unixTimestamp} (looking forward)`,
             );
 
-            dis.dispatch<ViewRoomPayload>({
-                action: Action.ViewRoom,
-                event_id: eventId,
-                highlighted: true,
-                room_id: roomId,
-                metricsTrigger: undefined, // room doesn't change
-            });
-        } catch (e) {
-            const code = e.errcode || e.statusCode;
-            // only show the dialog if failing for something other than a network error
-            // (e.g. no errcode or statusCode) as in that case the redactions end up in the
-            // detached queue and we show the room status bar to allow retry
-            if (typeof code !== "undefined") {
-                // display error message stating you couldn't delete this.
+            // Only try to navigate to the room if the user is still viewing the same
+            // room. We don't want to jump someone back to a room after a slow request
+            // if they've already navigated away to another room.
+            const currentRoomId = SdkContextClass.instance.roomViewStore.getRoomId();
+            if (currentRoomId === roomIdForJumpRequest) {
+                dispatcher.dispatch<ViewRoomPayload>({
+                    action: Action.ViewRoom,
+                    event_id: eventId,
+                    highlighted: true,
+                    room_id: roomIdForJumpRequest,
+                    metricsTrigger: undefined, // room doesn't change
+                });
+            } else {
+                logger.debug(
+                    `No longer navigating to date in room (jump to date) because the user already switched ` +
+                        `to another room: currentRoomId=${currentRoomId}, roomIdForJumpRequest=${roomIdForJumpRequest}`,
+                );
+            }
+        } catch (err) {
+            logger.error(
+                `Error occured while trying to find event in ${roomIdForJumpRequest} ` +
+                    `at timestamp=${unixTimestamp}:`,
+                err,
+            );
+
+            // Only display an error if the user is still viewing the same room. We
+            // don't want to worry someone about an error in a room they no longer care
+            // about after a slow request if they've already navigated away to another
+            // room.
+            const currentRoomId = SdkContextClass.instance.roomViewStore.getRoomId();
+            if (currentRoomId === roomIdForJumpRequest) {
+                let friendlyErrorMessage = `An error occured while trying to find and jump to the given date.`;
```

Comment:
Do we need backticks here? Just wondering if there's meant to be a JS variable in there.

Reply:
This is something that Prettier/eslint should yell about if we care.

This has been updated though ✅